app/src/main/java/org/simple/clinic/storage/monitoring/DbMethodLookup.kt

```diff
@@ -0,0 +1,46 @@
+package org.simple.clinic.storage.monitoring
+
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
+
+class DbMethodLookup @VisibleForTesting(otherwise = PRIVATE) constructor(
+    metadataCsv: String
+) {
+
+  private val allDaoMethods: List<DaoMetadata> = DaoMetadata.parse(metadataCsv)
+
+  fun find(daoName: String, lineNumber: Int): String? {
+    return allDaoMethods
+        .find { metadata -> metadata.matches(daoName, lineNumber) }
+        ?.methodName
+  }
+
+  private data class DaoMetadata(
+      val daoName: String,
+      val methodName: String,
+      val methodRange: IntRange
+  ) {
+
+    fun matches(daoName: String, lineNumber: Int): Boolean {
+      return this.daoName == daoName && lineNumber in methodRange
+    }
+
+    companion object {
+      fun parse(csv: String): List<DaoMetadata> {
+        return csv
+            .split('\n')
+            .asSequence()
```

Comment:
Any particular reason we converted it to sequence here?

Reply:
Not particularly, it was just an IDE recommendation that I auto-approved 😅 .